process/notifier/sovereignNotifier.go

```diff
@@ -101,7 +102,7 @@ func (notifier *sovereignNotifier) Notify(outportBlock *outport.OutportBlock) er
 		IncomingMiniBlocks: mbs,
 	}
 
-	notifier.notifyHandlers(extendedHeader)
+	notifier.notifyHandlers(outportBlock.BlockData.HeaderHash, extendedHeader)
```

Comment:
We need the hash of extendedHeader here so it must be calculated. outportBlock.BlockData.HeaderHash is the hash of shardHeader.

Reply:
Also we need to push the transactions included in the incoming mini blocks into sovereign node TxPool

Reply:
> We need the hash of extendedHeader here so it must be calculated. outportBlock.BlockData.HeaderHash is the hash of shardHeader. 

Right, missed it. Refactored as suggested + unit tests

> Also we need to push the transactions included in the incoming mini blocks into sovereign node TxPool

Very good point. Added task for it